File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        return help(root, null, null);  
    }

    private boolean help(TreeNode root, Integer min, Integer max)
    {
        if(root == null)
            return true;
        if (min != null && root.val <= min) 
            return false; 
        if (max != null && root.val >= max) 
            return false;    
        return help(root.left, min, root.val) && help(root.right, root.val, max);
    }
}

File: Data Structures & Algorithms/sqrtx/submission-0.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        res = 0 # משתנה שישמור את התוצאה הכי טובה שמצאנו עד כה
        
        while left <= right:
            mid = (left + right) // 2
            q = mid * mid
            
            if q == x:
                return mid 
            
            if q < x:
                res = mid 
                left = mid + 1
            else:
                right = mid - 1
                
        return res